source/constructs/api/data_source/service.py

```diff
@@ -2345,11 +2347,28 @@ def __delete_account(account_id: str, region: str):
         logger.error(traceback.format_exc())
 
 
-def query_glue_connections(account: AdminAccountInfo):
-    return __glue(account=account.account_id, region=account.region).get_connections(CatalogId=account.account_id,
-                                                                                     Filter={'ConnectionType': 'JDBC'},
-                                                                                     MaxResults=100,
-                                                                                     HidePassword=True)['ConnectionList']
+def query_glue_connections(account: AccountInfo):
+    res = []
+    account_id = account.account_id if account.account_provider_id == Provider.AWS_CLOUD.value else _admin_account_id
+    region = account.region if account.account_provider_id == Provider.AWS_CLOUD.value else _admin_account_region
+    list =  __glue(account=account_id, region=region).get_connections(CatalogId=account_id,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The API method `get_connections` returns paginated results instead of all results. Consider using the pagination API or checking one of the following keys in the response to verify that all results were returned: `IsTruncated`, `NextToken`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html)